src/engine.c

```diff
@@ -270,6 +276,26 @@ ucschar_strlen (const ucschar* str)
     return p - str;
 }
 
+static gboolean
+check_client_commit (const gchar *version)
+{
+    gboolean client_commit = FALSE;
+    gint len = 0;
+
+    gchar **versions = g_strsplit_set (version, ".", 3);
```

Comment:
The returned `versions` should be freed with `g_strfreev()` after use.

Reply:
Done